steps/s3/session.go

```diff
@@ -0,0 +1,179 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package s3steps
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/Telefonica/golium"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	aws_s "github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/aws/aws-sdk-go/service/s3/s3manager"
+	"github.com/sirupsen/logrus"
+)
+
+type Session struct {
+	Client           *aws_s.Session
+	CreatedDocuments []*CreatedDocument
+}
+
+type CreatedDocument struct {
+	bucket string
+	key    string
+}
+
+// newS3Session initiates a new aws session.
+func (s *Session) newS3Session(ctx context.Context) error {
+	logrus.Debugf("Creating a new S3 session")
+	var err error
+
+	s3Config := &aws.Config{
+		Region: aws.String("eu-west-1"),
+	}
+	// Check minio
+	minio := golium.Value(ctx, "[CONF:minio]").(bool)
+	if minio {
+		s3Config = &aws.Config{
+			Credentials:      credentials.NewStaticCredentials(golium.Value(ctx, "[CONF:minio-aws-access-key-id]").(string), golium.Value(ctx, "[CONF:minio-aws-secret-access-key]").(string), ""),
+			Endpoint:         aws.String(golium.Value(ctx, "[CONF:minio-endpoint]").(string)),
+			Region:           aws.String(golium.Value(ctx, "[CONF:minio-aws-region]").(string)),
+			DisableSSL:       aws.Bool(true),
+			S3ForcePathStyle: aws.Bool(true),
+		}
+	}
+
+	s.Client, err = aws_s.NewSession(s3Config)
+	if err != nil {
+		return fmt.Errorf("error creating s3 session. %v", err)
+	}
+	return nil
+}
+
+// uploadS3FileWithContent creates a new file in S3 with the content specified.
+func (s *Session) uploadS3FileWithContent(ctx context.Context, bucket, key, message string) error {
+	uploader := s3manager.NewUploader(s.Client)
+	_, err := uploader.Upload(&s3manager.UploadInput{
+		Bucket: aws.String(bucket),
+		Key:    aws.String(key),
+		Body:   strings.NewReader(message),
+	})
+	if err != nil {
+		return fmt.Errorf("unable to upload %q to %q, %v", key, bucket, err)
+	}
+
+	s.CreatedDocuments = append(s.CreatedDocuments, &CreatedDocument{bucket: bucket, key: key})
+	GetLogger().Log("upload", bucket, key)
+	return nil
+}
+
+// createBucket creates a new bucket.
+func (s *Session) createBucket(ctx context.Context, bucket string) error {
+	logrus.Debugf("Creating a new bucket: %s", bucket)
+	cparams := &s3.CreateBucketInput{
+		Bucket: aws.String(bucket),
+	}
+
+	s3Client := s3.New(s.Client)
+	_, err := s3Client.CreateBucket(cparams)
+	if err != nil {
+		logrus.Errorf("Error creating a new bucket: %s, err: %v", bucket, err)
+	}
+	return nil
+}
+
+// validateS3File checks the existence of a file in S3.
+func (s *Session) validateS3File(ctx context.Context, bucket, key string) error {
```

Comment:
Ain't be better to call this method `validateS3FileExists`?

Reply:
Good suggestion to be cleare:

```
- validateS3FileExists
- validateS3FileExistsWithContent
```

Reply:
Done